File: src/main/scala/com/liwenqiang/util/serdes/JsonSerializer.scala
package com.liwenqiang.util.serdes

import com.google.gson.Gson
import org.apache.kafka.common.serialization.Serializer

import java.nio.charset.Charset
import java.util

class JsonSerializer[Any] extends Serializer[Any]{

  private val gson:Gson = new Gson()

  override def configure(configs: util.Map[String, _], isKey: Boolean): Unit = {

  }

  override def serialize(topic: String, data: Any): Array[Byte] = {
    gson.toJson(data).getBytes(Charset.forName("UTF-8"))
  }

  override def close(): Unit = {

  }

}

File: src/main/scala/com/liwenqiang/util/serializer/JsonSerializer.java
package com.liwenqiang.util.serializer;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.liwenqiang.collectors.FixedSizePriorityQueue;
import org.apache.kafka.common.serialization.Serializer;

import java.nio.charset.StandardCharsets;
import java.util.Map;

public class JsonSerializer<T> implements Serializer<T> {

    private Gson gson;

    public JsonSerializer() {
        GsonBuilder builder = new GsonBuilder();
        builder.registerTypeAdapter(FixedSizePriorityQueue.class, new FixedSizePriorityQueueAdapter().nullSafe());
        gson = builder.create();
    }

    @Override
    public void configure(Map<String, ?> map, boolean b) {

    }

    @Override
    public byte[] serialize(String topic, T t) {
        return gson.toJson(t).getBytes(StandardCharsets.UTF_8);
    }

    @Override
    public void close() {

    }
}

